scripts/lib/links/ExternalLink.ts

```diff
@@ -57,3 +57,16 @@ export class ExternalLink {
     return `❌ ${error}. Appears in:\n${fileList.join("\n")}`;
   }
 }
+
+function getHeaders(link: string) {
+  const headers: HeadersInit = {
+    "User-Agent": "qiskit-documentation-broken-links-finder",
+  };
+
+  if (link.startsWith("https://github.com")) {
+    const ghToken = process.env.GITHUB_TOKEN;
+    headers["Authorization"] = ghToken ? `token ${ghToken}` : "";
```

Comment:
Setting the header to the empty string is different semantically than not setting it at all. I haven't tested if it actually changes end-behavior, but I think this expresses intent better

```suggestion
    const ghToken = process.env.GITHUB_TOKEN;
    if (ghToken) {
      headers["Authorization"] = `token ${ghToken}`;
    }
```